lib/Dialect/SPIRV/SPIRVDialect.cpp

```diff
@@ -53,6 +53,18 @@ SPIRVDialect::SPIRVDialect(MLIRContext *context)
 // Type Parsing
 //===----------------------------------------------------------------------===//
 
+// Forward declarations.
+template <typename ValTy>
```

Comment:
Could you forward-declare these as `static` to be consistent with later? I'm seeing compiler errors of "static declaration of 'parseAndVerify' follows non-static declaration".

Reply:
@antiagainst oh, sorry about this, I've updated it, this is indeed a bug regarding to the standard, but I cannot reproduce an error on my local machine with gcc 8, even with reduced test case for template functions.

Reply:
No worries! And thanks for the ping regarding LLVM C++14 update! :)